packages/devtools/lib/src/timeline/frames_bar_plotly.dart

```diff
@@ -305,3 +345,136 @@ class FramesBarPlotly {
     legendClick(domName, f);
   }
 }
+
+class SelectTrace {
+  SelectTrace(this.traceIndex, this.ptNumber, this.xValue, this.yValue);
+
+  final int traceIndex;
+  final int ptNumber;
+  final int xValue;
+  final num yValue;
+}
+
+class Selection {
+  Selection(this._domName, dynamic graphDiv)
+      : _data = getProperty(graphDiv, 'data');
+
+  final String _domName;
+  final List<Data> _data;
+  List<SelectTrace> selectInfo = [];
+
+  bool isSelected(List<int> xValues) =>
+      selectInfo.length == 2 &&
+      selectInfo[0].xValue == xValues[0] &&
+      selectInfo[1].xValue == xValues[1];
+
+  int get selectedPointNumber =>
+      selectInfo.isNotEmpty ? selectInfo[0].ptNumber : -1;
+
+  void select(List<int> pointNumbers, List<int> traces, List<int> xValues,
+      List<num> yValues) {
+    // Supports one bar selection and not selecting a currently selected bar.
+    assert(traces.length == 2 &&
```

Comment:
nit: perhaps make 2 a const so that we know what it means and can use it here and elsewhere

Reply:
It means two traces per data point.  I'll add a constant.